src/backend/app/Console/Commands/ResendRejectedMessages.php

```diff
@@ -24,32 +28,83 @@ class ResendRejectedMessages extends AbstractSharedCommand {
      *
      * @return void
      */
-    public function __construct(private Sms $sms) {
+    public function __construct(
+        private Sms $sms,
+        private SmsGatewayResolverService $gatewayResolver,
+    ) {
         parent::__construct();
     }
 
     public function handle(): void {
+        if (!$this->gatewayResolver->hasActiveProvider()) {
+            $this->error('No active SMS provider configured. Please enable an SMS provider plugin in the settings.');
+            Log::error('Resend rejected messages command failed: No active SMS provider configured');
+
+            return;
+        }
+
         $amountToSend = (int) $this->argument('amount');
         $this->sms
             ->where('direction', 1)
             ->where('status', -1)
             ->orderBy('id')
             ->take($amountToSend)
-            ->get()->each(function ($sms) {
-                Log::info("Resending rejected message {$sms->id} to {$sms->receiver}");
-                $smsAndroidSettings = SmsAndroidSetting::getResponsible();
-                $callback = sprintf($smsAndroidSettings->callback, $sms->uuid);
-                $sms->status = 0;
-                $sms->gateway_id = $smsAndroidSettings->getId();
-                $sms->save();
-
-                resolve(AndroidGateway::class)
-                    ->sendSms(
-                        $sms->receiver,
-                        $sms->body,
-                        $callback,
-                        $smsAndroidSettings
-                    );
+            ->get()->each(function (Sms $sms) {
+                try {
+                    Log::info("Resending rejected message {$sms->id} to {$sms->receiver}");
+                    [$gateway, $viberId] = $this->gatewayResolver->determineGateway($sms->receiver);
+
+                    match ($gateway) {
+                        SmsGatewayResolverService::VIBER_GATEWAY => $this->sendViaViber($sms, $viberId),
+                        SmsGatewayResolverService::AFRICAS_TALKING_GATEWAY => $this->sendViaAfricasTalking($sms),
+                        default => $this->sendViaAndroid($sms),
+                    };
+                } catch (NoActiveSmsProviderException $exception) {
+                    Log::error("Failed to resend message {$sms->id}: {$exception->getMessage()}");
+                    $this->error("Failed to resend message {$sms->id}: No active SMS provider");
+                }
             });
     }
+
+    private function sendViaViber(Sms $sms, ?string $viberId): void {
+        if ($viberId) {
+            resolve(ViberGateway::class)
+                ->sendSms(
+                    $sms->body,
+                    $viberId,
+                    $sms
+                );
+            $sms->status = 0;
+            $sms->gateway_id = MpmPlugin::VIBER_MESSAGING;
+            $sms->save();
+        }
+    }
```

Comment:
It seems like a lot of this code is already present in `SmsSender`: https://github.com/EnAccess/micropowermanager/blob/16d9b67b2e4a2fd76848b50af952c5c968fc5545/src/backend/app/Sms/Senders/SmsSender.php#L19-L39

I don't fully understand how it's supposed to be used, but is there a way we can reduce redundancy and duplication here, and somehow re-use (or at least consolidate) some of the existing code?